deployment/keystone/changeset/deploy_ocr3_test.go

```diff
@@ -37,3 +44,109 @@ func TestDeployOCR3(t *testing.T) {
 	oaddrs, _ := resp.AddressBook.AddressesForChain(env.AllChainSelectors()[1])
 	assert.Len(t, oaddrs, 0)
 }
+
+func TestConfigureOCR3(t *testing.T) {
+	t.Parallel()
+	lggr := logger.Test(t)
+	//ctx := tests.Context(t)
+
+	c := kslib.OracleConfigWithSecrets{
+		OracleConfig: kslib.OracleConfig{
+			MaxFaultyOracles:    1,
+			DeltaProgressMillis: 12345,
+		},
+		OCRSecrets: deployment.XXXGenerateTestOCRSecrets(),
+	}
+
+	t.Run("no mcms", func(t *testing.T) {
+
+		te := Setup(t, TestConfig{
+			WFDonConfig:     DonConfig{N: 4},
+			AssetDonConfig:  DonConfig{N: 4},
+			WriterDonConfig: DonConfig{N: 4},
+			NumChains:       1,
+		})
+
+		var wfNodes []string
+		for id, _ := range te.WFNodes {
+			wfNodes = append(wfNodes, id)
+		}
+
+		w := &bytes.Buffer{}
+		cfg := changeset.ConfigureOCR3Config{
+			ChainSel:             te.RegistrySelector,
+			NodeIDs:              wfNodes,
+			OCR3Config:           &c,
+			WriteGeneratedConfig: w,
+			UseMCMS:              false,
+		}
+
+		csOut, err := changeset.ConfigureOCR3Contract(te.Env, cfg)
+		require.NoError(t, err)
+		var got kslib.OCR2OracleConfig
+		err = json.Unmarshal(w.Bytes(), &got)
+		require.NoError(t, err)
+		assert.Len(t, got.Signers, 4)
+		assert.Len(t, got.Transmitters, 4)
+		assert.Nil(t, csOut.Proposals)
+	})
+
+	t.Run("mcms", func(t *testing.T) {
+		te := Setup(t, TestConfig{
+			WFDonConfig:     DonConfig{N: 4},
+			AssetDonConfig:  DonConfig{N: 4},
+			WriterDonConfig: DonConfig{N: 4},
+			NumChains:       1,
+			EnableMCMS:      true,
+		})
+
+		var wfNodes []string
+		for id, _ := range te.WFNodes {
+			wfNodes = append(wfNodes, id)
+		}
+
+		w := &bytes.Buffer{}
+		cfg := changeset.ConfigureOCR3Config{
+			ChainSel:             te.RegistrySelector,
+			NodeIDs:              wfNodes,
+			OCR3Config:           &c,
+			WriteGeneratedConfig: w,
+			UseMCMS:              true,
+		}
+
+		csOut, err := changeset.ConfigureOCR3Contract(te.Env, cfg)
+		require.NoError(t, err)
+		var got kslib.OCR2OracleConfig
+		err = json.Unmarshal(w.Bytes(), &got)
+		require.NoError(t, err)
+		assert.Len(t, got.Signers, 4)
+		assert.Len(t, got.Transmitters, 4)
+		assert.NotNil(t, csOut.Proposals)
+		t.Logf("got: %v", csOut.Proposals[0])
+
+		contractSetsResp, err := kslib.GetContractSets(lggr, &kslib.GetContractSetsRequest{
+			Chains:      te.Env.Chains,
+			AddressBook: te.Env.ExistingAddresses,
+		})
+		require.NoError(t, err)
+		var timelocks = map[uint64]*gethwrappers.RBACTimelock{
+			te.RegistrySelector: contractSetsResp.ContractSets[te.RegistrySelector].Timelock,
+		}
+		// now apply the changeset such that the proposal is signed and execed
+		w2 := &bytes.Buffer{}
+		cfg.WriteGeneratedConfig = w2
+		_, err = commonchangeset.ApplyChangesets(t, te.Env, timelocks, []commonchangeset.ChangesetApplication{
+			{
+				Changeset: commonchangeset.WrapChangeSet(changeset.ConfigureOCR3Contract),
+				Config:    cfg,
+			},
+		})
+		if err != nil {
```

Comment:
this fails with `failed to apply changeset: failed to execute proposal: execution reverted: RBACTimelock: underlying transaction reverted`